partner_category_restrict/models/partner.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2023 Akretion (https://www.akretion.com).
+# @author Pierrick Brun <pierrick.brun@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.constrains("category_id")
+    def _check_category_restrict_filter(self):
+        for record in self:
+            for categ in record.category_id.filtered("restrict_filter_id"):
+                restrict_domain = categ.restrict_filter_id.sudo()._get_eval_domain()
+                if not self.env["res.partner"].search_count(
```

Comment:
```suggestion
                if not self.search_count(
```

Reply:
I prefer using the explicit way tbh.

Also I've already had problems with calling this kind of function with self being a non empty recordset. The function used self as if it was empty and it created weird issues